src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -238,7 +272,91 @@ private void validateSsl() {
     }
   }
 
+  private void validateConnection() {
+    try {
+      if (!client.ping(RequestOptions.DEFAULT)) {
+        throw new IOException("Cannot connect to Elasticsearch.");
+      }
+    } catch (IOException e) {
+      String errorMessage = String.format(
+          "Could not connect to Elasticsearch due to %s",
+          e.getMessage()
+      );
+      addErrorMessage(CONNECTION_URL_CONFIG, errorMessage);
+
+      if (config.isAuthenticatedConnection()) {
+        errorMessage = String.format(
+            "Could not authenticate the user. Check the '%s' and '%s'. Error message: %s.",
+            CONNECTION_USERNAME_CONFIG,
+            CONNECTION_PASSWORD_CONFIG,
+            e.getMessage()
+        );
+        addErrorMessage(CONNECTION_USERNAME_CONFIG, errorMessage);
+        addErrorMessage(CONNECTION_PASSWORD_CONFIG, errorMessage);
+      }
+
+      if (config.isSslEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your SSL settings. Error message: %s",
```

Comment:
nit: which `SSL` configs? Give the user a prefix maybe, which configs should he check?

Reply:
this error message is added for every single config that it affects below

Reply:
https://github.com/confluentinc/kafka-connect-elasticsearch/pull/498/files/34da65886dbbe4dea395a9ec9f7584eb2c825ca2#diff-01aa6b506fb570054df3884b36841d52284ab0a07ac84ce1767da88987b6a2c8R304